src/substructures/smarts.jl

```diff
@@ -0,0 +1,33 @@
+using MolecularGraph: 
+    QueryMol, 
+    smartstomol, substructmatches
+export SMARTSQuery
+
+struct SMARTSQuery
+    query::String
+    query_graph::QueryMol
+
+    function SMARTSQuery(query::String)
+        new(query, smartstomol(query))
+    end
+end
+
+function _to_substructure(mol, m; adjacent_bonds=false)
+    matched_atoms = keys(m)
+
+    atom_view = filter(:number => n -> n ∈ matched_atoms, mol.atoms, view=true)
+    bond_view = filter(
+        [:a1, a2] => (a1, a2) -> 
+            adjacent_bonds ? a1 ∈ matched_atoms && a2 ∈ matched_atoms
+                           : a1 ∈ matched_atoms || a2 ∈ matched_atoms,
+        mol.bonds, view=true)
+
+    Substructure("$(query) on $(mol.name)", atom_view, bond_view)
```

Comment:
`ERROR: UndefVarError: query not defined`

Reply:
Also, the constructor call requires additional arguments (parent molecule and properties).

Reply:
good catch. that broke during the rebase. should be fixed now